valgrind.supp

```diff
@@ -0,0 +1,965 @@
+   fun:_ZN5boost3log11v2_mt_posix3aux11record_pumpINS1_7sources18severity_logger_mtIN2ml4core7CLogger6ELevelEEEED2Ev
+   fun:_ZN5boost3log11v2_mt_posix3aux11record_pumpINS1_7sources18severity_logger_mtIN2ml4core7CLogger6ELevelEEEED1Ev
+   fun:_ZN2ml5maths7CSignal3fftERNSt3__16vectorINS2_7complexIdEENS2_9allocatorIS5_EEEE
+   fun:_ZN2ml5maths7CSignal16autocorrelationsERKNSt3__16vectorINS0_16CBasicStatistics21SSampleCentralMomentsINS_4core13CFloatStorageELj1EEENS2_9allocatorIS8_EEEERNS3_IdNS9_IdEEEE
+}
+{
+   <insert_a_suppression_name_here>
+   Memcheck:Cond
+   fun:_ZNKSt3__115basic_stringbufIcNS_11char_traitsIcEENS_9allocatorIcEEE3strEv
+   fun:_ZNKSt3__119basic_ostringstreamIcNS_11char_traitsIcEENS_9allocatorIcEEE3strEv
+   fun:_ZNSt3__1lsIdcNS_11char_traitsIcEEEERNS_13basic_ostreamIT0_T1_EES7_RKNS_7complexIT_EE
+   fun:_ZN2ml4core14printer_detail12CLeafPrinterIN5boost17integral_constantIbLb0EEEE6print_INSt3__17complexIdEEEENS8_12basic_stringIcNS8_11char_traitsIcEENS8_9allocatorIcEEEERKT_S5_
+   fun:_ZN2ml4core14printer_detail12CLeafPrinterIN5boost17integral_constantIbLb0EEEE5printINSt3__17complexIdEEEENS8_12basic_stringIcNS8_11char_traitsIcEENS8_9allocatorIcEEEERKT_
+   fun:_ZN2ml4core14printer_detail12CNodePrinterIN5boost17integral_constantIbLb0EEENS0_17CContainerPrinterEE5printINSt3__17complexIdEEEENS9_12basic_stringIcNS9_11char_traitsIcEENS9_9allocatorIcEEEERKT_
+   fun:_ZN2ml4core17CContainerPrinter12printElementINSt3__17complexIdEEEENS3_12basic_stringIcNS3_11char_traitsIcEENS3_9allocatorIcEEEERKT_
+   fun:_ZN2ml4core17CContainerPrinter5printINSt3__111__wrap_iterIPKNS3_7complexIdEEEEEENS3_12basic_stringIcNS3_11char_traitsIcEENS3_9allocatorIcEEEET_SG_
+   fun:_ZN2ml4core17CContainerPrinter5printINSt3__16vectorINS3_7complexIdEENS3_9allocatorIS6_EEEEEENS3_12basic_stringIcNS3_11char_traitsIcEENS7_IcEEEERKT_
+   fun:_ZN2ml5maths7CSignal3fftERNSt3__16vectorINS2_7complexIdEENS2_9allocatorIS5_EEEE
+   fun:_ZN2ml5maths7CSignal16autocorrelationsERKNSt3__16vectorINS0_16CBasicStatistics21SSampleCentralMomentsINS_4core13CFloatStorageELj1EEENS2_9allocatorIS8_EEEERNS3_IdNS9_IdEEEE
+   fun:_ZN2ml5maths12_GLOBAL__N_132mostSignificantPeriodicComponentENSt3__16vectorINS0_16CBasicStatistics21SSampleCentralMomentsINS_4core13CFloatStorageELj1EEENS2_9allocatorIS8_EEEE
+}
+{
+   <insert_a_suppression_name_here>
+   Memcheck:Value8
+   fun:_ZNSt3__115basic_streambufIcNS_11char_traitsIcEEE5sputcEc
+   fun:_ZNSt3__115basic_stringbufIcNS_11char_traitsIcEENS_9allocatorIcEEE8overflowEi
+   fun:_ZNSt3__115basic_streambufIcNS_11char_traitsIcEEE6xsputnEPKcl
+   fun:_ZNSt3__115basic_streambufIcNS_11char_traitsIcEEE5sputnEPKcl
+   fun:_ZNSt3__116__pad_and_outputIcNS_11char_traitsIcEEEENS_19ostreambuf_iteratorIT_T0_EES6_PKS4_S8_S8_RNS_8ios_baseES4_
+   fun:_ZNSt3__124__put_character_sequenceIcNS_11char_traitsIcEEEERNS_13basic_ostreamIT_T0_EES7_PKS4_m
+   fun:_ZNSt3__1lsIcNS_11char_traitsIcEENS_9allocatorIcEEEERNS_13basic_ostreamIT_T0_EES9_RKNS_12basic_stringIS6_S7_T1_EE
```

Comment:
Probably a good rule of thumb would be to end each stack trace at the line containing `basic_ostream`, then delete the resulting duplicates.